Refactor templates to project sections

zglueck opened the pull request:
This pull request stems from a conversation in #45. The logic inside the `_default/list.html` template was starting to get more and more complex with requirements for additional front matter in the markdown files.

This pull request utilizes Hugo's template lookup functionality which will match the section name (android, java, web) to a folder in the layouts directory with the appropriate `single.html` and `list.html` templates. There is some duplication of html code between the projects, but the reduction in front matter variables and template logic as well as the ease of customization from project to project makes this change a step in the right direction.

zglueck committed:
Refactor list templates to sections
- Remove default list template logic
- Add debuging indicators
- Remove front matter which is no longer required

zglueck committed:
Remove debug statements and update comments

zglueck committed:
Remove Developer Guide link

zglueck committed:
Remove unused template

zglueck committed:
Remove Developer Guide links

zglueck committed:
Remove debug code

pdavidc commented:
Thanks, @zglueck. Is it possible to refactor each project's navigation links into a partial template?

zglueck committed:
Refactor out navigation and list to partials
- Move the project specific navigation html elements to partials to reduce code duplication
- Move the panel list to a partial to reduce code duplication

zglueck commented:
@pdavidc refactored nav skipped my mind!

0be0361704e5891f1 has the refactored templates. Now each navigation bar is in a partial and labeled as <project>-navigation.html

I went just a little bit further and moved the listing html elements (shared between java, android, and web) to a partial as well. The move really cleans up the list.html templates.

zglueck committed:
Fix indentation

zglueck committed:
Refactor panel list template for more generality
- Pull containing div and content out

pdavidc reviewed (approved):
Looks great!
